File: Fase3_compila/test/t11OkGer.pas
program t11;

var
	apple, pear, orange : integer;
	text : string;
	number : real;
	a, b : char;

	procedure writentimes( text: string; n : integer; );
	var
		i : integer;
	
	begin
		i := 1;
		while i <= n do
			write(text);
		endwhile;
	end;

	function factorial( n : integer; ) : integer;
	var
		i, product : integer;
	
	begin
		product := 1;
		i := 2;
		while i <= n do
			product := product * i;
		endwhile;
	end;

begin
	apple := 5;
	a := "A";
	pear := factorial (apple);
	orange := 10;
	number := 4.2;
	text := "Hello, world!";
	write(text);
	read(text);
	writeln(text);
end.

File: Fase3_compila/test/t11OkGer.c
#include <stdio.h>
#include <string.h>

int apple, pear, orange;
string text;
float number;
char a, b;

void writentimes( string text, int n ) {
	int i;
	i = 1;

	while ( i <= n ) {
		printf("%s", text);
	}
}

int factorial( int n ) {
	int i;
	int product;

	product = 1;
	i = 2;
	
	while ( i <= n ) {
		product = product * i;
	}
}

int main() {
	apple = 5;
	a = 'A';
	pear = factorial(apple);
	orange = 10;
	number = 4.2;
	text = "Hello, world!";
	printf("%s", text);
	gets(text);
	printf("%s\n", text);
}
